scripts/exregional_run_observer.sh

```diff
@@ -0,0 +1,545 @@
+    cp_vrfy  ${bkpath}/${restart_prefix}fv_tracer.res.tile1.nc           fv3_tracer
+    cp_vrfy  ${bkpath}/${restart_prefix}sfc_data.nc                      fv3_sfcdata
+    cp_vrfy  ${bkpath}/${restart_prefix}coupler.res                      coupler.res
+  else
+    print_err_msg_exit "$VERBOSE" "Error: cannot find background: ${checkfile}"
+  fi
+  fv3lam_bg_type=0
+fi
+
+#
+# Build namelist and run GSI
+#
+#-----------------------------------------------------------------------
+# Link the AMV bufr file
+ifsatbufr=.false.
+
+# Build the GSI namelist on-the-fly
+. ${fixgriddir}/gsiparm.anl.sh
+cat << EOF > gsiparm.anl
+$gsi_namelist
+EOF
+
+#
+#-----------------------------------------------------------------------
+#
+# Run the GSI. 
+#
+#-----------------------------------------------------------------------
+#
+$APRUN ./gsi.x < gsiparm.anl > stdout 2>&1 || print_err_msg_exit "\
+Call to executable to run GSI returned with nonzero exit code."
+#
+ls -l * > list_run_directory_mem000${ensmem}
+#
+mv stdout stdout_mem000${ensmem}
+mv gsiparm.anl gsiparm.anl_mem000${ensmem}
+rm -f pe*.obs_setup
+
+#-----------------------------------------------------------------------
+# Loop over first and last outer loops to generate innovation
+# diagnostic files for indicated observation types (groups)
+#
+# NOTE:  Since we set miter=2 in GSI namelist SETUP, outer
+#        loop 03 will contain innovations with respect to 
+#        the analysis.  Creation of o-a innovation files
+#        is triggered by write_diag(3)=.true.  The setting
+#        write_diag(1)=.true. turns on creation of o-g
+#        innovation files.
+#-----------------------------------------------------------------------
+#
+
+netcdf_diag=${netcdf_diag:-".false."}
+binary_diag=${binary_diag:-".true."}
+
+loops="01"
+for loop in $loops; do
+
+case $loop in
+  01) string=ges;;
+  03) string=anl;;
+   *) string=$loop;;
+esac
+
+#  Collect diagnostic files for obs types (groups) below
+if [ $binary_diag = ".true." ]; then
+   listall=`ls pe* | cut -f2 -d"." | awk '{print substr($0, 0, length($0)-3)}' | sort | uniq `
+   for type in $listall; do
+      count=`ls pe*.${type}_${loop} | wc -l`
+      if [[ $count -gt 0 ]]; then
+         `cat pe*.${type}_${loop} > diag_${type}_${string}.mem000${ensmem}`
+      fi
+   done
+fi
+
+done
+
+rm -f fv3_dynvars fv3_tracer fv3_sfcdata
+
+# next member
+   (( ensmem += 1 ))
+
+done
+
+# Delete/Unlink unneeded files
+rm -f obs*
+rm -f pe*
+rm -f gsi.x
+rm -f fv3_grid_spec berror_stats satbias* gfs_data.tile7.halo0.nc_b
+find *.bin -maxdepth 1 -exec unlink '{}' \;
+
+#
+#-----------------------------------------------------------------------
+#
+# Print message indicating successful completion of script.
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "
+========================================================================
+ANALYSIS GSI completed successfully!!!
```

Comment:
Suggest changing to "GSI observer completed successfully".

Reply:
Changed and pushed the commit. Thanks!